spec/takeaway_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'takeaway'
+
+describe Takeaway do
+  subject(:takeaway) { described_class.new }
+  let(:dish) { double :dish => "Dish 1"}
+  let(:order) { double :order, inject: [["Dish 1" => 2.00]], length: 3}
+  let(:quantity) { double :quantity => 3}
+  let(:total) { double :total => 6}
+
+  describe '#menu' do
+    it 'should display a list of dishes with prices' do
+      expect(takeaway.list).not_to be_empty
+    end
+  end
+
+  describe '#choose' do
+    it 'should enable a user to select a dish' do
+      takeaway.choose("Dish 1")
+      expect(takeaway.order).to include(["Dish 1", 1.0])
+    end
+
+    it 'should enable a user to select multiple quantities of dishes' do
+      takeaway.choose("Dish 1", 2)
+      expect(takeaway.order).to include(["Dish 1", 1.0])
+    end
+  end
+
+  describe '#place_order' do
+
+    it 'raises an error if the total cost does not equal the sum of dishes' do
+      allow(order).to receive(:each)
+      takeaway.calculate_quantities(order)
+      takeaway.calculate_cost(order)
+      expect{ takeaway.place_order(order, quantity, total) }.to raise_error "Wrong total!"
```

Comment:
Metrics/LineLength: Line is too long. [90/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)